packages/tailwindcss-react-aria-components/src/index.js

```diff
@@ -5,37 +5,60 @@ const attributes = {
     ['hover', 'hovered'],
     ['focus', 'focused'],
     'focus-visible',
+    'focus-within',
     'pressed',
     'disabled',
     'drop-target',
     'dragging',
     'empty',
+    'allows-dragging',
     'allows-removing',
     'allows-sorting',
     'placeholder',
     'selected',
     'indeterminate',
-    'readonly',
+    ['read-only', 'readonly'],
     'required',
     'entering',
     'exiting',
     'open',
     'unavailable',
+    'outside-month',
+    'outside-visible-range',
+    'selection-start',
+    'selection-end',
     'current',
-    'invalid'
+    'invalid',
+    'resizing'
   ],
   enum: {
     placement: ['left', 'right', 'top', 'bottom'],
     type: ['literal', 'year', 'month', 'day'],
     layout: ['grid', 'stack'],
-    orientation: ['horizontal', 'vertical']
+    orientation: ['horizontal', 'vertical'],
+    'selection-mode': ['single', 'multiple'],
+    'resizable-direction': ['right', 'left', 'both'],
+    'sort-direction': ['ascending', 'descending']
   }
 };
 
+const shortNames = {
+  'selection-mode': 'selection',
+  'resizable-direction': 'resizable',
+  'sort-direction': 'sort'
+};
+
 // Variants we use that are already defined by Tailwind:
 // https://github.com/tailwindlabs/tailwindcss/blob/a2fa6932767ab328515f743d6188c2164ad2a5de/src/corePlugins.js#L84
-const nativeVariants = ['indeterminate', 'required', 'invalid', 'empty', 'focus-visible', 'disabled'];
-const nativeVariantSelectors = new Map([...nativeVariants.map((variant) => [variant, `:${variant}`]), ['hovered', ':hover'], ['focused', ':focus'], ['open', '[open]']]);
+const nativeVariants = ['indeterminate', 'required', 'invalid', 'empty', 'focus-visible', 'focus-within', 'disabled'];
+const nativeVariantSelectors = new Map([
+  ...nativeVariants.map((variant) => [variant, `:${variant}`]),
+  ['hovered', ':hover'],
+  ['focused', ':focus'],
+  ['readonly', ':read-only'],
+  ['open', '[open]'],
+  ['placeholder', '::placeholder']
```

Comment:
The `placeholder:` modifier that is built into tailwind actually targets a pseudo element not a pseudo class. There is also `placeholder-shown:` which is a bit closer. Should we match that or keep `placeholder:`?

Reply:
Since we're matching tailwind's built-in variants for the others, my vote would be to use `placeholder-shown`